libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -161,6 +161,37 @@ def _should_bind_tools(model: LanguageModelLike, tools: Sequence[BaseTool]) -> b
     return False
 
 
+def _validate_messages(
+    messages: Sequence[BaseMessage],
+) -> None:
+    """Validate that all tool calls in AIMessages have a corresponding ToolMessage."""
+    all_tool_calls = [
+        tool_call
+        for message in messages
+        if isinstance(message, AIMessage)
+        for tool_call in message.tool_calls
+    ]
+    answered_tool_call_ids = {
```

Comment:
(nit) it's internal so doesn't super matter, but `answered` and `unanswered` are slightly confusing terms. 

Some better names would probably be: Unacknowledged/acknowledged

There's probably something even better